app/controllers/concerns/application_activities_concern.rb

```diff
@@ -0,0 +1,9 @@
+module ApplicationActivitiesConcern
+  extend ActiveSupport::Concern
+
+  def recent_activities(course, number)
+    return if number < 0 || !number.is_a?(Integer)
+    PublicActivity::Activity.where(owner: course).where.not('key like ?', '%notification').
```

Comment:
If they are the same then what is the point of a database schema? You cannot shove values in willy-nilly.

I agree with your definition of trackable.

I am not convinced your decision to assign owners and recipients is the best available. How do you reconcile things like activity feeds with popups?